daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/RepParsers.scala

```diff
@@ -249,7 +249,7 @@ class RepUnboundedParser(occursCountKind: OccursCountKind.Value, rParser: Parser
     var priorState: PState.Mark = null
     var markLeakCausedByException = false
 
-    Assert.invariant(initialState.processorStatus eq Success)
+    if (initialState.processorStatus ne Success) return
```

Comment:
Want to avoid returns. Scala implements them as throws which are relatively heavy. Can we instead just wrap the call to parseAllRepeats with the if(initialState.processorStatus ne Success) check?

Also, can you explain why this happens? It seems if the initial state of a RepParser is not Success, we never should have enetered the RepParser in the first place. Maybe there's a bug elsewhere?